librz/analysis/block.c

```diff
@@ -1041,29 +1048,35 @@ RZ_API bool rz_analysis_block_set_op_offset(RzAnalysisBlock *block, size_t i, ut
 }
 
 /**
- * @return the address of the instruction that occupies a given offset or UT64_MAX if off is not in the block.
+ * \return the index of the instruction that occupies a given offset or -1 if addr is not in the block.
  */
-RZ_API ut64 rz_analysis_block_get_op_addr_in(RzAnalysisBlock *bb, ut64 off) {
-	ut16 delta, delta_off, last_delta;
-	int i;
-
-	if (!rz_analysis_block_contains(bb, off)) {
-		return UT64_MAX;
+RZ_API int rz_analysis_block_get_op_index_in(RzAnalysisBlock *bb, ut64 addr) {
+	if (!rz_analysis_block_contains(bb, addr)) {
+		return -1;
 	}
-	last_delta = 0;
-	delta_off = off - bb->addr;
-	for (i = 0; i < bb->ninstr; i++) {
-		delta = rz_analysis_block_get_op_offset(bb, i);
+	ut16 delta_off = addr - bb->addr;
+	for (int i = 0; i < bb->ninstr; i++) {
+		ut16 delta = rz_analysis_block_get_op_offset(bb, i);
 		if (delta > delta_off) {
-			return bb->addr + last_delta;
+			return i - 1;
 		}
-		last_delta = delta;
 	}
-	return bb->addr + last_delta;
+	return bb->ninstr - 1;
 }
 
 /**
- * @return the size of the i-th instruction in a basic block
+ * \return the address of the instruction that occupies a given offset or UT64_MAX if off is not in the block.
+ */
+RZ_API ut64 rz_analysis_block_get_op_addr_in(RzAnalysisBlock *bb, ut64 off) {
```

Comment:
Maybe you could rename the variable to `offset` and then use `\p offset` in the Doxygen description?

Reply:
Will change to `\p addr` too